lib/pbench/server/database/alembic/versions/f628657bed56_user_table_update_oidc.py

```diff
@@ -0,0 +1,117 @@
+"""Update User table to only keep username, oidc_id, roles of the user
+
+Revision ID: f628657bed56
+Revises: fa12f45a2a5a
+Create Date: 2023-02-26 23:24:16.650879
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "f628657bed56"
+down_revision = "9cc638cb865c"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.add_column("users", sa.Column("_roles", sa.String(length=255), nullable=True))
+    op.execute("ALTER TABLE users DROP CONSTRAINT users_pkey CASCADE")
+    op.alter_column(
+        "users",
+        "id",
+        existing_type=sa.INTEGER(),
+        type_=sa.String(length=255),
+        existing_nullable=False,
+    )
+    op.alter_column(
+        "users", "username", existing_type=sa.VARCHAR(length=255), nullable=False
+    )
```

Comment:
> Why do we pass existing_nullable in the first call

`existing_nullable` operation indicates whether a column that already exists in the database can be set to allow null values.

The `existing_nullable` argument comes into play when you want to modify an existing column that already has data in it. If we set `existing_nullable=True`, this means that the column will be set to allow null values if it doesn't already, and any existing data in the column will not be affected. If we set `existing_nullable=False`, this means that the column will be set to disallow null values if it currently allows them and any existing data that is null will cause an error. This is because the column is no longer allowed to have null values, so the existing null values are now considered invalid data. _I think_ that's what we want for the `id` column (as its a primary key) and we never expect it to have a `null` value if it does then it should result in an error. (Now, when we deploy this we would create a new users table so the scenario wouldn't matter but I think this is a good check to have.) 

> but not the second

`existing_nullable` does not exist in the `username` column because I think for the `username` column Alembic does not need to change the existing nullability of the column. In other words, if the column currently allows null values, it will continue to allow null values after the migration, and if it currently disallows null values, it will continue to disallow null values.

> why do we pass nullable in the second but not the first

We aren't specifying `nullable=False` because we added a primary key constraint on `id` column so its not needed.

Reply:
Thanks for the explanation!

So, in the first case we don't need to specify `nullable=False` (and so we don't) because that's implied by the fact that the `id` column already is the primary key.  And, on that basis, the `existing_nullable=False` specification should not find any issues, but perhaps it's prudent to have it look.

The second case is odd, because there's no change in the source code to this column...so why do we need a migration alteration for it?  (It's already a non-nullable `String(255)`.)

Reply:
That's a good point. I have no idea why `alembic` decided to add this line in the migration file, I'll have to do some digging.

Reply:
I did some testing around this and alembic doesn't seem to complain when I remove this alteration, nor it has any problem building tables. So I guess can remove it.

Reply:
Have you tried regenerating this code?  (I'm guessing that the regeneration would omit this alteration, now.)

I assume that the Alembic "check" test would catch any divergence between a migration script generated early in a PR's evolution and subsequent incompatible changes in the PR, but...as a best practice, we should probably generate the migration script late in the PR process (or, perhaps, _re_-generate it periodically during review...if there are any relevant changes requested).